src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -784,3 +814,21 @@ async function getSortedViaTrips(
   };
   return sortedData;
 }
+
+function createPublicCodeLineMap(lines: LineFragment[]) {
+  const publicCodeLineList = new Map<string, string[]>();
+  lines.forEach((line) => {
+    if (line.publicCode) {
+      if (publicCodeLineList.has(line.publicCode)) {
+        publicCodeLineList.get(line.publicCode)?.push(line.id);
+      } else {
+        publicCodeLineList.set(line.publicCode, [line.id]);
+      }
+    }
+  });
+
+  // @ts-ignore
+  // @ts-expect-error: This is needed to allow iteration through Map objects
+  // @ts-nocheck
+  return JSON.stringify([...publicCodeLineList]);
```

Comment:
I think this will be simplified by using just something like:
```suggestion
function createPublicCodeLineMap(lines: LineFragment[]) {
  const publicCodeLineList: Record<string, string[]> = {};
  for (const line of lines) {
    if (!line.publicCode) continue;
    if (!publicCodeLineList[line.publicCode]) {
      publicCodeLineList[line.publicCode] = [];
    }

    publicCodeLineList[line.publicCode].push(line.id);
  }
  return publicCodeLineList;
```